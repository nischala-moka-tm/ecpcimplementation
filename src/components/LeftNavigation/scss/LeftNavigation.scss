.nav {
  .nav-item {
    &:first-child {
      a {
        &::before {
          background-image: url("../../../assets/dashboard-default.svg");
          width: 15px;
          height: 15px;
          left: 10px;
        }

        &.active {
          &::before {
            background-image: url("../../../assets/dashboard-select.svg");
          }
        }
      }
    }

    a {
      letter-spacing: -0.01px;
      color: #000000;
      font-size: 14px;
      line-height: 16px;
      background: #cccccc 0% 0% no-repeat padding-box;
      padding: 15px 5px 15px 35px;
      margin-bottom: 5px;
      position: relative;
      text-decoration: none;

      &::before {
        content: "";
        background-image: url("../../../assets/inactive-icon.PNG");
        width: 20px;
        height: 20px;
        position: absolute;
        top: 50%;
        transform: translateY(-50%);
        left: 8px;
        background-size: 100% 100%;
        background-repeat: no-repeat;
      }

      &:hover,
      &:focus {
        text-decoration: none;
        color: inherit;
      }

      &.active {
        background: #000000 0% 0% no-repeat padding-box;
        color: #fff;
        text-decoration: none;

        &::before {
          background-image: url("../../../assets/Icon-material-select-all.svg");
        }
      }

      &.right-arrow {
        position: relative;

        &::after {
          position: absolute;
          content: "";
          border-left: 5px solid #000;
          border-top: 5px solid transparent;
          border-bottom: 5px solid transparent;
          margin-left: 5px;
          top: 20px;
          right: 15px;
        }

        &.open {
          &::after {
            transform: rotate(90deg);
          }
        }
      }
    }


  }

  .sub-dropdown {
    display: block;

    .nav-item {
      a {
        &::before {
          content: none;
          background: none;
        }
      }
    }
  }
}