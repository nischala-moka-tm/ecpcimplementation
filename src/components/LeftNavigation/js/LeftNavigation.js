import React, { useState } from "react";
import { Nav } from "react-bootstrap";
import "../scss/LeftNavigation.scss";

function LeftNavigation(props) {
  const [SubC1, setSubC1] = useState(false);
  const [isActive, setActive] = useState(false);
  const handleToggle = () => {
    setActive(!isActive);
    setSubC1(!SubC1);
  };
  return (
    <Nav
      defaultActiveKey={`#${
        window.location.pathname.split("/")[2]
          ? window.location.pathname.split("/")[2]
          : "dashboard"
      }`}
      className="flex-column"
    >
      <Nav.Item>
        <Nav.Link
          href="#dashboard"
          onClick={() => {
            props.onClickTab(`${props.path}/dashboard`);
            setSubC1(false);
            setActive(false);
          }}
        >
          Dashboard
        </Nav.Link>
      </Nav.Item>
      <Nav.Item>
        <Nav.Link
          href="#manage"
          onClick={() => {
            props.onClickTab(`${props.path}/manage`);
            setSubC1(false);
            setActive(false);
          }}
        >
          Manage Permissions/Preferences
        </Nav.Link>
      </Nav.Item>
      <Nav.Item>
        <Nav.Link
          className={isActive ? "right-arrow open" : "right-arrow"}
          onClick={handleToggle}
        >
          Reports
        </Nav.Link>
      </Nav.Item>
      {SubC1 && (
        <Nav className="sub-dropdown">
          <Nav.Item>
            <Nav.Link
              eventKey="report-1"
              href="#report-perm"
              onClick={() => props.onClickTab(`${props.path}/report-perm`)}
            >
              Permission/Preference
            </Nav.Link>
          </Nav.Item>
          <Nav.Item>
            <Nav.Link
              eventKey="report-2"
              href="#report-audit"
              onClick={() => props.onClickTab(`${props.path}/report-audit`)}
            >
              Audit Trail
            </Nav.Link>
          </Nav.Item>
        </Nav>
      )}
    </Nav>
  );
}

export default LeftNavigation;
