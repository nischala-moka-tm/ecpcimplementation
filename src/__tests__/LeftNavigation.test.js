import React from "react";
import Enzyme, { mount } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import LeftNavigation from "../components/LeftNavigation/js/LeftNavigation";
import renderer from "react-test-renderer";
import { act } from "react-dom/test-utils";
import { render, fireEvent } from "@testing-library/react";

Enzyme.configure({ adapter: new Adapter() });
const onClickTab = (childPath) => {
  console.log("path");
};
const path = "/dashboard-admin";
it("Rendering Left Navigation", () => {
  const wrapper = mount(<LeftNavigation path={path} onClickTab={onClickTab} />);
  expect(wrapper.find(".flex-column")).toBeDefined();
  for (var index; index <= wrapper.find(".nav-link").length; index++) {
    wrapper.find(".nav-link").at(index).simulate("click");
  }
});

it("Rendering Left Navigation", () => {
  const wrapper = render(
    <LeftNavigation path={path} onClickTab={onClickTab} />
  );
  fireEvent.click(wrapper.getByText("Dashboard"));
  fireEvent.click(wrapper.getByText("Manage Permissions/Preferences"));
});

it("renders correctly when there are no items", () => {
  const tree = renderer
    .create(<LeftNavigation path={path} onClickTab={onClickTab} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

describe("renders <LeftNavigation />", () => {
  it("should pass mount snapshot test", async () => {
    let wrapper;
    await act(async () => {
      wrapper = mount(<LeftNavigation path={path} onClickTab={onClickTab} />);
    });
    expect(wrapper).toMatchSnapshot();
  });
});
