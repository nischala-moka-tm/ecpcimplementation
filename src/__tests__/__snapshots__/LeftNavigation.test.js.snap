// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <LeftNavigation /> should pass mount snapshot test 1`] = `ReactWrapper {}`;

exports[`renders correctly when there are no items 1`] = `
<div
  className="flex-column nav"
  onKeyDown={[Function]}
>
  <div
    className="emptyNav"
  />
  <div
    className="nav-item"
  >
    <a
      className="nav-link active"
      data-rr-ui-event-key="#dashboard"
      disabled={false}
      href="#dashboard"
      onClick={[Function]}
    >
      Dashboard
    </a>
  </div>
  <div
    className="nav-item"
  >
    <a
      className="nav-link"
      data-rr-ui-event-key="#manage"
      disabled={false}
      href="#manage"
      onClick={[Function]}
    >
      Manage Permissions/Preferences
    </a>
  </div>
  <div
    className="nav-item"
  >
    <a
      className="right-arrow nav-link"
      data-rr-ui-event-key={null}
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Reports
    </a>
  </div>
  <div
    className="emptySecondNav"
  />
</div>
`;
